### PR TITLE
feat: optional import of decord, fix for sentence feature extractor

### PR DESCRIPTION
**Pull request by jarulraj**
👋 Thanks for submitting a Pull Request to EvaDB!

🙌 We want to make contributing to EvaDB as easy and transparent as possible. Here are a few tips to get you started:

- 🔍 Search existing EvaDB [PRs](https://github.com/georgia-tech-db/eva/pulls) to see if a similar PR already exists.
- 🔗 Link this PR to a EvaDB [issue](https://github.com/georgia-tech-db/eva/issues) to help us understand what bug fix or feature is being implemented.
- 📈 Provide before and after profiling results to help us quantify the improvement your PR provides (if applicable).

👉 Please see our ✅ [Contributing Guide](https://evadb.readthedocs.io/en/stable/source/contribute/index.html) for more details.

### CODE REVIEW

File: evadb/udfs/sentence_feature_extractor.py [hunk 1]
```diff
@@ -32,6 +32,7 @@ def setup(self):
             "sentence-transformers/all-MiniLM-L6-v2"
         )
         self.model = AutoModel.from_pretrained("sentence-transformers/all-MiniLM-L6-v2")
+        self.device = None
```
Comment (jiashenC) [inline evadb/udfs/sentence_feature_extractor.py:35]: Can you remind me why do we need this var?

File: evadb/udfs/sentence_feature_extractor.py [hunk 2]
```diff
@@ -32,6 +32,7 @@ def setup(self):
             "sentence-transformers/all-MiniLM-L6-v2"
```
Comment (jiashenC) [inline evadb/udfs/sentence_feature_extractor.py:32]: Do we converge on which UDF to keep this or the `sentence_transformer_feature_extractor`? I think they are the same model, but the other implementation looks cleaner than this.

File: evadb/udfs/sentence_feature_extractor.py (hunk 1 above)
Comment (jarulraj) [reply evadb/udfs/sentence_feature_extractor.py]: We need to set `self.model_device` -- not sure if `self.device` also needs to be set.

Comment (jarulraj) [reply evadb/udfs/sentence_feature_extractor.py]: On a machine with only a CPU, we get this error -- `has no attribute` `model_device`

File: evadb/udfs/sentence_feature_extractor.py (hunk 2 above)
Comment (jarulraj) [reply evadb/udfs/sentence_feature_extractor.py:32]: Which one did you add?

Comment (jarulraj) [reply evadb/udfs/sentence_feature_extractor.py:32]: Let's just call this `text_feature_extractor`?

Comment (jiashenC) [reply evadb/udfs/sentence_feature_extractor.py:32]: Sure. I added this one, but i think other one is much simpler and also the device issue will disappear. I just found that this UDF’s implementation has a bug.

**Pull request merged**